### DISCUSSION TITLE
FR: Cropping

### DISCUSSION THREAD

**Discussion by Skeletorjus** [sca075/mqtt_vacuum_camera#15] [Ideas]
Since the map often includes lots of empty space it would be useful to be able to crop the image. This could either be done by specifying pixels to crop from top/bottom/left/right or by drawing a square with a cursor akin to a zone-creator.

The output _can_ be cropped in [xiaomi-vacuum-map-card](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card), but that won't affect the generated snapshots or any other visualization of the map.

**Comment by sca075** [comment]
The Valetudo maps are normally size wise are (5120 x 5120 or even bigger) huge the camera image can be actually crop on the options (default is 50% but you can use less for example 25% to reduce it).
![Screenshot 2023-08-16 at 11 04 06](https://github.com/sca075/valetudo_vacuum_camera/assets/82227818/d21a98f8-8846-466c-94ce-932368e60f1d)
I would add a image trim function on the camera (same as the one the card actually offer) to select top, bottom, left or right space to be "cut out" from the resulting image.. this would help to centre the image without using the cropping or trimming function on the card and also keep the snapshots same as the output on the card.

**Comment by sca075** [comment]
F.Y.I. trim function implemented and tested on the test code I use for troubleshooting, will be available next Friday on the new version of this integration hopefully, I still need to work out the new config_flow, where the new parameters can be set. I actually thought to add also a text on the top right conner of the image that will output the vacuum status (this will be also included as an option) so that when the camera take the snapshot the status of the vacuum will be display also on the image. I count to close also the story of the room configuration on the new version but no promises here. Additionally from our discussions it comes out I think a great idea.. would be really cool if Gen1 vacuums get the zones predefined (same principle of the rooms, I think this should be possible by following the walls). This last two actually needs a little of lines to add :) but in the end I guess can make the real difference.

**Comment by sca075** [comment]
@Skeletorjus hope everything works fine with your vacuums, I just released the v1.3.2 that support the images trimming. Thanks a lots for your inputs much appreciated 🥇

**Reply by Skeletorjus** [reply]
Hi! Just testing the cropping, but I can't get a handle of how it's supposed to work. 
I first set up my camera with a image crop for 25% (a setting of 0 crashes the integration by the way) and open at the resulting image in an image editor to find out the size of the area i want to show. I then want to remove the 600 top pixels, 250 bottom pixels, 150 left pixels and 250 right pixels. I then enter those values in the config flow, but I can't seem to achieve the result I expect. I'm probably misunderstanding something here..?

Thank you again for working on this!

**Reply by Skeletorjus** [reply]
I just relalized that top/bottom and left/right are the opposite of what I thought!  "Trim crop top area" removes pixels from the bottom and vice versa. Same with left/right. When I flipped all the values it worked as expected! Are the options simply labelled wrong or should they be renamed "Pixels to remove from top" or simething similar to avoid confusion?

**Reply by sca075** [reply]
@Skeletorjus  thanks a lot man, I experienced something similar too.. may I ask you what is the rotation value you use? probably the issue is that when we rotate the image this values rotate with the image.. if it is so.. (just for reference I used 90 deg. on the tests I made) it can be an easy fix to do.. do you mind to open an issue for please?

**Reply by sca075** [reply]
@Skeletorjus let me say.. you are a really valuable user here.. I found the problem.. you are right the values change with the rotation of the image. This will be fixed soon ;) thanks a lot.. (N.B. Im opening an issue to keep track of it)

**Reply by Skeletorjus** [reply]
Thank a lot! Glad you found the issue. I use a rotation of zero.

**Reply by sca075** [reply]
@Skeletorjus the function is fixed and v1.3.3 is released.. I hope this can help and hat you can enjoy it :)

**Reply by Skeletorjus** [reply]
Working great! Thanks!

**Reply by sca075** [reply]
@Skeletorjus Thank to you for the really good advice:) happy that it did work 😃
I'm closing this discussion at this point, also because it is everything documented on the docs. 
Thank you once again ;)
